frontend/src/AccountDetailsTile.tsx

```diff
@@ -55,26 +66,45 @@ export class AccountDetailsTile extends React.Component<AccountDetailsProps, Acc
                 } else {
                     response.json().then(data => {
                         this.setState({
-                            eventBus: this.state.eventBus,
-                            account: data
+                            account: this.mergeCredentials(data),
+                            displayNameLastSaved: data.displayName,
+                            platformIdLastSaved: data.platformId
                         });
                     });
+                    this.props.eventBus.fireEvent(EventBusEventType.REFRESH_SOCIAL_MEDIA_ACCOUNTS);
                 }
             });
     }
 
+    /** Merges the credentials of the new account with the unsaved ones to avoid losing unsaved changes. 
+     * This would happen because the account is refreshed on save. 
+     * Refreshes the selected account afterwards.*/
+    private mergeCredentials(account: SocialMediaAccount): SocialMediaAccount {
+        this.state.account.credentials.forEach(c => {
+            if (account.credentials.filter(x => x.descriptor === c.descriptor).length === 0) {
```

Comment:
`(account.credentials.filter(x => x.descriptor === c.descriptor).length === 0` is the same as
```suggestion
            if (account.credentials.find(x => x.descriptor === c.descriptor) == null) {
```

Reply:
Would it be beneficial to use a [`Map`](https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Map) for these descriptor array-of-object things? Instead of `array.find(x => x.descriptor === searchValue) == null` you could do `!map.has(searchValue)`. And merging would be as easy as `new Map(...oldDescriptorMap, ...newDescriptorMap)`. What do you think?

Reply:
This would definitely be cleaner, although the performance impact should not matter at our current object counts.